packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts

```diff
@@ -420,6 +435,329 @@ export function mikroOrmBaseRepositoryFactory<T extends object = object>(
+          const originalEntry = originalDataMap.get((entityEntry as any).id)!
+          const reconstructedEntry = reconstructedResponse[i]
+
+          await promiseAll(
+            allRelations?.map(async (relation) => {
+              const relationName = relation.name as keyof T
+              if (!upsertConfig.relationsToUpsert?.includes(relationName)) {
+                return
+              }
+
+              // TODO: Handle ONE_TO_ONE
+              // One to one and Many to one are handled outside of the assignment as they need to happen before the main entity is created
+              if (
+                relation.reference === ReferenceType.ONE_TO_ONE ||
+                relation.reference === ReferenceType.MANY_TO_ONE
+              ) {
+                return
+              }
+
+              reconstructedEntry[relationName] =
+                await this.assignCollectionRelation_(
+                  manager,
+                  { ...originalEntry, id: (entityEntry as any).id },
+                  relation
+                )
+              return
+            })
+          )
+        })
+      )
+
+      // // We want to populate the identity map with the data that was written to the DB, and return an entity object
+      // return reconstructedResponse.map((r) =>
+      //   manager.create(entity, r, { persist: false })
+      // )
+
+      return reconstructedResponse
+    }
+
+    // FUTURE: We can make this performant by only aggregating the operations, but only executing them at the end.
+    protected async assignCollectionRelation_(
+      manager: SqlEntityManager,
+      data: T,
+      relation: EntityProperty
+    ) {
+      const dataForRelation = data[relation.name]
+      // If the field is not set, we ignore it. Null and empty arrays are a valid input and are handled below
+      if (dataForRelation === undefined) {
+        return undefined
+      }
+
+      // Make sure the data is correctly initialized with IDs before using it
+      const normalizedData = dataForRelation.map((normalizedItem) => {
+        return this.getEntityWithId(manager, relation.type, normalizedItem)
+      })
+
+      if (relation.reference === ReferenceType.MANY_TO_MANY) {
+        const currentPivotColumn = relation.inverseJoinColumns[0]
+        const parentPivotColumn = relation.joinColumns[0]
+
+        if (!normalizedData.length) {
+          await manager.nativeDelete(relation.pivotEntity, {
+            [parentPivotColumn]: (data as any).id,
+          })
+
+          return normalizedData
+        }
+
+        await this.upsertMany_(manager, relation.type, normalizedData)
+
+        const pivotData = normalizedData.map((currModel) => {
+          return {
+            [parentPivotColumn]: (data as any).id,
+            [currentPivotColumn]: currModel.id,
+          }
+        })
+
+        const qb = manager.qb(relation.pivotEntity)
+        await qb.insert(pivotData).onConflict().ignore().execute()
```

Comment:
question: Do we want to always ignore conflicts on insertion?

Reply:
@olivermrbl since the primary key in the pivot table will be the IDs of the two sides, a conflict means that the relation is already there so we can ignore the conflict. 

Of course, if we support composite keys, or other custom properties this might need to change.

Reply:
Got it thanks